drivers/led_strip/apa102.c

```diff
@@ -21,7 +21,18 @@ static int apa102_update(const struct device *dev, void *buf, size_t size)
 {
 	const struct apa102_config *config = dev->config;
 	static const uint8_t zeros[] = { 0, 0, 0, 0 };
-	static const uint8_t ones[] = { 0xFF, 0xFF, 0xFF, 0xFF };
+
+	/*
+	 * the only function of the “End frame” is to supply more clock pulses
+	 * to the string until the data has permeated to the last LED. The
+	 * number of clock pulses required is exactly half the total number
+	 * of LEDs in the string
+	 */
+	BUILD_ASSERT(sizeof(struct led_rgb) == 4);
+	uint8_t ones[((size / sizeof(struct led_rgb)) / sizeof(uint8_t) / 2) + 1];
+
+	memset(ones, 0xFF, sizeof(ones));
```

Comment:
Is there a bug with the current code ?

Reply:
yes, not enough `1` are sent in case there are more than 64 LEDs in the strip. The following LED after the 64th are not actuated.

It's not correctly documented in the APA102 datasheet

Reply:
Understood. Please fill a bug report and submit a separate PR to fix it.

Reply:
Hi @fouge. Are you moving forward on this ?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/76533